src/components/LoginRegister/RegisterInput.js

```diff
@@ -0,0 +1,75 @@
+import React, { useState } from 'react';
+import { useApolloClient } from '@apollo/client';
+import { useDispatch } from 'react-redux';
+import { Flex, Input, Button, Heading, VStack } from '@chakra-ui/react';
+import { register } from '../../data/actions/auth';
+import { Redirect } from 'react-router-dom';
+
+const RegisterInput = () => {
+  const client = useApolloClient();
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [submitState, setSubmitState] = useState('');
+  const dispatch = useDispatch();
+
+  const handleClick = async e => {
+    e.preventDefault();
+    dispatch(register(firstName, lastName, email, password, client)).then(
+      registered => {
+        if (!registered) {
+          setSubmitState('FAIL');
+        } else {
+          // Redirect to home page after registering
+          return <Redirect to={{ pathname: '/home' }} />;
+        }
```

Comment:
we shouldn't actually need this `Redirect` due to you updating `REGISTER_SUCCESS` correctly in the Redux reducer and L45 in `App.js` (Redirects away from login page to home page if the `user` object is not null)

```suggestion
        if (!registered) {
          setSubmitState('FAIL');
        }
```

Reply:
Removed!